### ISSUE TITLE
[BUG] - Channel join

### ISSUE THREAD

**Issue by RodolfoSilva** [braverhealth/phoenix-socket-dart#66]
`channel.join()` only works when called inside openStream, if it is called outside some times this does not work.

✅ Works
```dart
final socket = PhoenixSocket(endpoint);
socket.connect();
final channel = socket.addChannel(topic: "any", parameters: {"r": "test"});
channel.messages.listen((event) {
  print("This will work");
});
socket.openStream.listen((event) {
  channel.join();
});
```

⛔️ Doesn't work
```dart
final socket = PhoenixSocket(endpoint);
socket.connect();
final channel = socket.addChannel(topic: "any", parameters: {"r": "test"});
channel.messages.listen((event) {
  print("This will be lost");
});
channel.join();
```

**Comment by RodolfoSilva** [comment]
@matehat, do you have any advice about the situation?

**Comment by RodolfoSilva** [comment]
@matehat, do you have any advice about the situation?

**Comment by matehat** [comment]
Investigating now

**Comment by matehat** [comment]
Fix is in the 0.7.2 release

**Comment by RodolfoSilva** [comment]
Thanks @matehat
